packages/core/src/amazonqFeatureDev/session/sessionState.ts

```diff
@@ -284,29 +289,46 @@ export class CodeGenState extends CodeGenBase implements SessionState {
     async interact(action: SessionStateAction): Promise<SessionStateInteraction> {
         return telemetry.amazonq_codeGenerationInvoke.run(async (span) => {
             try {
+                action.tokenSource?.token.onCancellationRequested(() => {
+                    this.isCancellationRequested = true
+                    if (action.tokenSource) {
+                        this.tokenSource = action.tokenSource
+                    }
+                })
                 span.record({
                     amazonqConversationId: this.config.conversationId,
                     credentialStartUrl: AuthUtil.instance.startUrl,
                 })
 
+                action.tokenSource?.token.onCancellationRequested(() => {
```

Comment:
Why are lies 292-297 and this the exact same?

Reply:
Good call, rebased / merge issue. Just fixing this one